### PR TITLE
feat!: remove --credentials arg in favor of Application Default Credentials

### PR DESCRIPTION
**Pull request by tbpg**
We should use ADC (e.g. `GOOGLE_APPLICATION_CREDENTIALS`) instead.

Alternatively, we could keep the argument, but this PR would have to add
code rather than mostly delete code.

Once `docuploader.credentials.find` makes `credentials_file` optional, we can remove that arg.

### CODE REVIEW

**Comment by jskeet** [comment]
I'm fine with using ADC by default, but I don't think it's a good idea to remove the *option* of a credentials file.

When performing a release build for example, we run our integration tests with one credential but use a different one for uploading docs - this change would make that more complicated, unless I've misunderstood something.

**Comment by tbpg** [comment]
`docpipeline` is only run in two situations:
* Devs developing locally.
* Automated builds, configured in this repo.

So, I'm not too concerned about support for the `--credentials` argument here. If you disagree, I'm happy to work it back in -- would be a little more refactoring, but not too bad.

**Comment by jskeet** [comment]
Ah, for docpipeline I'm absolutely fine with this. Was getting confused with docuploader. Apologies.

File: docpipeline/generate.py [hunk 1]
```diff
@@ -311,7 +309,7 @@ def version_sort(v):
     return semver.VersionInfo.parse(v)
 
 
-def build_blobs(client, blobs, credentials):
+def build_blobs(client, blobs, credentials, project_id):
```
Comment (jskeet) [inline docpipeline/generate.py:312]: We don't seem to be using `project_id` at the moment. Do we actually need it?

**Review by jskeet — APPROVED** [review]
One nit, and with the caveat that I'm not a Python code, but generally looks fine.

**Comment by tbpg** [comment]
It ended up being very straightforward to re-add support for the flag. So, I went ahead and did it.

At least now commands in your terminal history will continue to work. :smile:

**Comment by tbpg** [comment]
> It ended up being very straightforward to re-add support for the flag. So, I went ahead and did it.

Just kidding. I forgot we were passing the credentials through to the `docuploader` command. So, I reverted the commit.

File: docpipeline/generate.py (hunk 1 above)
Comment (tbpg) [reply docpipeline/generate.py]: Good catch! Done. We also didn't need `credentials`, so I removed that, too.

**Pull request merged**